lib/github-pages-health-check/domain.rb

```diff
@@ -100,7 +100,7 @@ def initialize(host, nameservers: :default)
 
         @host = normalize_host(host)
         @nameservers = nameservers
-        @resolver = GitHubPages::HealthCheck::Resolver.new(host,
+        @resolver = GitHubPages::HealthCheck::Resolver.new(@host,
```

Comment:
Found this while undertaking some more testing just then when I managed to trigger some further Dnsruby exceptions when using a Unicode encoded IDN. Seems we weren't passing the normalised host through to the `Resolver` class.

Reply:
The `host` method should have been normalized?

Reply:
As `host` exists as a locally scoped variable in the `initialize` method, my understanding is that we need to explicitly refer to the instance level scoped `host` where we want to use it, otherwise we will get back the locally scoped one, which is pre `normalize_host`.

```
$ cat local-vs-instance.rb
def test(host)
  @host = host.upcase
  puts "host: #{host}, @host: #{@host}"
end

test("github.com")

$ ruby local-vs-instance.rb
host: github.com, @host: GITHUB.COM
```

Reply:
Totally right! The `host` parameter is first in the order of precedence, then the interpreter will attempt to find a method called `host`. I'd love to use `self.host` here so that we're calling the method. I'm worried about reordering these calls and causing `@host` to be `nil` by accident. What do you think?

Reply:
Thanks @parkr, I'm a little lost here, as from what I understand we don't have a `host` method?, would you mind expanding on what you mean here?

Reply:
On line 6, there's an `attr_reader` for `:host` which creates a reader for `@host`! That's where I'm understanding that there is a host method. 😄

Reply:
Thanks @parkr, it finally clicked 😄 I've updated the PR to use `self.host`.